src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
+
+  let sum = 0;
+
+  children.forEach((ch) => sum += ch.born - people.find(
```

Comment:
You can use a more readable coding style or optimize code 

```suggestion
children.forEach((ch) => {
  return sum += ch.born - people.find((m) => {
    return m.name === ch.mother;
      }).born;
});
```